pkg/kube/krt/status_test.go

```diff
@@ -101,6 +107,20 @@ func TestCollectionStatus(t *testing.T) {
 	tt.Empty()
 	assert.Equal(t, Status.GetKey("namespace/svc").Status, SimpleServiceStatus{TotalEndpoints: 1})
 
+	// Remove status entirely
+	pod.Labels["delete-status"] = "true"
+	pc.UpdateStatus(pod)
+	assert.EventuallyEqual(t, fetcherSorted(SimpleEndpoints), []SimpleEndpoint{{pod.Name, svc.Name, pod.Namespace, "1.2.3.5"}})
+	// There should NOT be an event caused by the change, as there is no change to report.
+	tt.WaitOrdered("delete/namespace/svc")
+	assert.Equal(t, Status.GetKey("namespace/svc"), nil)
+
+	// add status back
+	pod.Labels["delete-status"] = "false"
+	pc.UpdateStatus(pod)
+	tt.WaitOrdered("add/namespace/svc")
+	assert.Equal(t, Status.GetKey("namespace/svc").Status, SimpleServiceStatus{TotalEndpoints: 1})
```

Comment:
as stated in https://github.com/istio/istio/pull/55924#discussion_r2044411809, I think we should add another test case: where a collection *unselects* (returns nil for both status and the tranformed object) an object that was previously selected and had a status